samples/extensions/descriptor_buffer_basic/descriptor_buffer_basic.cpp

```diff
@@ -0,0 +1,471 @@
+		vkDestroyDescriptorSetLayout(get_device().get_handle(), descriptor_set_layout_buffer, nullptr);
+		vkDestroyDescriptorSetLayout(get_device().get_handle(), descriptor_set_layout_image, nullptr);
+		for (auto &cube : cubes)
+		{
+			cube.uniform_buffer.reset();
+			cube.texture.image.reset();
+			vkDestroySampler(get_device().get_handle(), cube.texture.sampler, nullptr);
+		}
+		uniform_buffers.scene.reset();
+		resource_descriptor_buffer.reset();
+		image_descriptor_buffer.reset();
+	}
+}
+
+void DescriptorBufferBasic::request_gpu_features(vkb::PhysicalDevice &gpu)
+{
+	// Enable anisotropic filtering if supported
+	if (gpu.get_features().samplerAnisotropy)
+	{
+		gpu.get_mutable_requested_features().samplerAnisotropy = VK_TRUE;
+	}
+
+	// Enable features required for this example
+
+	// We need device addresses for buffers in certain places
+	auto &requested_buffer_device_address_features               = gpu.request_extension_features<VkPhysicalDeviceBufferDeviceAddressFeatures>(VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_BUFFER_DEVICE_ADDRESS_FEATURES);
+	requested_buffer_device_address_features.bufferDeviceAddress = VK_TRUE;
+
+	// We need to enable the descriptor buffer feature of the VK_EXT_descriptor_buffer extension
+	auto &requested_buffer_descriptor_buffer_features            = gpu.request_extension_features<VkPhysicalDeviceDescriptorBufferFeaturesEXT>(VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_DESCRIPTOR_BUFFER_FEATURES_EXT);
+	requested_buffer_descriptor_buffer_features.descriptorBuffer = VK_TRUE;
+}
+
+void DescriptorBufferBasic::build_command_buffers()
+{
+	VkCommandBufferBeginInfo command_buffer_begin_info = vkb::initializers::command_buffer_begin_info();
+
+	VkClearValue clear_values[2];
+	clear_values[0].color        = default_clear_color;
+	clear_values[1].depthStencil = {0.0f, 0};
+
+	VkRenderPassBeginInfo render_pass_begin_info    = vkb::initializers::render_pass_begin_info();
+	render_pass_begin_info.renderPass               = render_pass;
+	render_pass_begin_info.renderArea.offset.x      = 0;
+	render_pass_begin_info.renderArea.offset.y      = 0;
+	render_pass_begin_info.renderArea.extent.width  = width;
+	render_pass_begin_info.renderArea.extent.height = height;
+	render_pass_begin_info.clearValueCount          = 2;
+	render_pass_begin_info.pClearValues             = clear_values;
+
+	for (int32_t i = 0; i < draw_cmd_buffers.size(); ++i)
+	{
+		render_pass_begin_info.framebuffer = framebuffers[i];
+
+		VK_CHECK(vkBeginCommandBuffer(draw_cmd_buffers[i], &command_buffer_begin_info));
+
+		vkCmdBeginRenderPass(draw_cmd_buffers[i], &render_pass_begin_info, VK_SUBPASS_CONTENTS_INLINE);
+
+		vkCmdBindPipeline(draw_cmd_buffers[i], VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline);
+
+		VkViewport viewport = vkb::initializers::viewport(static_cast<float>(width), static_cast<float>(height), 0.0f, 1.0f);
+		vkCmdSetViewport(draw_cmd_buffers[i], 0, 1, &viewport);
+
+		VkRect2D scissor = vkb::initializers::rect2D(width, height, 0, 0);
```

Comment:
width/height are unsigned, rect2d takes singed.  Can we static_cast to int for both here?

Reply:
I'd prefer that solution. Doing it in the framework would be the cleaner way.